jetty-ee10/jetty-ee10-websocket/jetty-ee10-websocket-jetty-server/src/main/java/org/eclipse/jetty/ee10/websocket/server/JettyWebSocketServerContainer.java

```diff
@@ -203,12 +207,15 @@ public boolean upgrade(JettyWebSocketCreator creator, HttpServletRequest request
             try
             {
                 Object webSocket = creator.createWebSocket(new DelegatedServerUpgradeRequest(req), new DelegatedServerUpgradeResponse(resp));
-                cb.succeeded();
+                if (webSocket == null)
+                    cb.succeeded();
                 return webSocket;
             }
             catch (Throwable t)
             {
-                cb.failed(t);
+                if (LOG.isDebugEnabled())
+                    LOG.debug("Could not create WebSocket endpoint", t);
+                Response.writeError(req, resp, cb, HttpStatus.INTERNAL_SERVER_ERROR_500, "Could not create WebSocket endpoint");
```

Comment:
Can you check whether `cb.failed()` is enough, because the javadocs of `creator.createWebSocket()` say to use `JettyServerUpgradeResponse. sendError()`.

Reply:
I think `cb.failed()` is probably better. This will result in the `JettyWebSocketServerContainer#upgrade` throwing, which can be caught and handled by their application, or if its not caught will result in a `sendError()` anyway.